.github/workflows/trivy.yml

```diff
@@ -6,49 +6,108 @@ name: Trivy scan
-    paths:
-      - '**/go.mod'
-      - '**/pom.xml'
-      - '**/*gradle.lockfile'
-      - '**/*.sbt.lock'
-      - '**/requirements.txt'
-      - '**/poetry.lock'
-      - '**/Pipfile.lock'
-      - '**/*trivy*'
 
 # Declare default permissions as nothing.
 permissions: {}
 
 jobs:
-  build:
-    name: Build
-    runs-on: ubuntu-20.04
+  update-trivy-db:
+    name: Update Trivy DB
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code to see if we run on default (master) branch
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
+
+      - name: Skip update on not default branch
+        if: github.ref != format('refs/heads/{0}', github.event.repository.default_branch)
+        run: |
+          echo "Not on the default branch"
+          exit 0
+
+      - name: Setup oras
+        if: github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
+        uses: oras-project/setup-oras@9c92598691bfef1424de2f8fae81941568f5889c  # v1.21
+
+      - name: Get current date to create cache entry
+        if: github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
+        id: date
+        run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+
+      - name: Check if Trivy DB cached already
+        if: github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
+        id: db-downloaded
+        uses: actions/cache/restore@6849a6489940f00c2f30c0fb92c6274307ccb58a  # v4.1.2
+        with:
+          path: ${{ github.workspace }}/.cache/trivy
+          key: cache-trivy-${{ steps.date.outputs.date }}
+          lookup-only: true
+
+      - name: Download and extract the vulnerability DB
+        if: |
+          steps.db-downloaded.outputs.cache-hit != 'true' &&
+          github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/.cache/trivy/db
+          oras pull ghcr.io/aquasecurity/trivy-db:2
+          tar -xzf db.tar.gz -C $GITHUB_WORKSPACE/.cache/trivy/db
+          rm db.tar.gz
+
+      - name: Download and extract the Java DB
+        if: |
+          steps.db-downloaded.outputs.cache-hit != 'true' &&
+          github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/.cache/trivy/java-db
+          oras pull ghcr.io/aquasecurity/trivy-java-db:1
+          tar -xzf javadb.tar.gz -C $GITHUB_WORKSPACE/.cache/trivy/java-db
+          rm javadb.tar.gz
+
+      - name: Save DBs in cache of the default (master) branch
+        if: |
+          steps.db-downloaded.outputs.cache-hit != 'true' &&
+          github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
+        uses: actions/cache/save@6849a6489940f00c2f30c0fb92c6274307ccb58a  # v4.1.2
+        with:
+          path: ${{ github.workspace }}/.cache/trivy
+          key: cache-trivy-${{ steps.date.outputs.date }}
```

Comment:
Is this subject to races?  I.e. what if two CI runs on the default branch both end up here at the same time?  Do they corrupt each other's (same) cache entry?